PyriteCore/src/display/texture.h

```diff
@@ -77,6 +80,33 @@ struct Cubemap
   ID3D11ShaderResourceView *m_texture;
 };
 
+struct TextureArray
+{
+
+    size_t m_width, m_height;
+    ID3D11Resource* m_resource;
+    ID3D11ShaderResourceView* m_textureArray;
+    std::vector<Texture> m_textures;
+    size_t m_arrayCount;
+    bool m_isCubeArray = false;
+
+public:
+
+    TextureArray(size_t width, size_t height, size_t count = 8, bool bIsDepthOnly = false, bool bIsCubeArray = false);
+    ~TextureArray();
+    static void CopyToTextureArray(const std::vector<Texture>& textures, TextureArray& outArray);
+    static void CopyToTextureArray(const std::vector<Cubemap>& cubes, TextureArray& outArray);
+    
+    ID3D11ShaderResourceView* getRawTexture()   const { return m_textureArray; }
+    ID3D11Resource* getRawResource()            const { return m_resource; }
+    size_t getWidth()                           const { return m_width; }
+    size_t getHeight()                          const { return m_height; }
+    size_t getArrayCount()                      const { return m_arrayCount; }
```

Comment:
That getter is misleading for cubemap texture arrays, it should probably return the same value that was passed to the constructor but it will return 6x that